packages/crypto/src/ed25519.rs

```diff
@@ -126,48 +100,18 @@ pub fn ed25519_batch_verify(
     debug_assert_eq!(messages.len(), signatures_len);
     debug_assert_eq!(messages.len(), public_keys.len());
 
-    let mut batch = ed25519::batch::Verifier::new();
+    let mut batch = batch::Verifier::new();
 
-    for (((i, &message), &signature), &public_key) in
+    for (((_i, &message), &signature), &public_key) in
         (1..).zip(&messages).zip(signatures).zip(&public_keys)
```

Comment:
The alternative is
```
    for ((&message, &signature), &public_key) in messages
        .iter()
        .zip(signatures.iter())
        .zip(public_keys.iter()
```
Choose your poison.

Reply:
I'm not familiar with the zip API and just did the least invasive change possible to your code. I'll try what you suggest, thanks

Reply:
Oh, I think the way it stands now is probably better / clearer. Just wanted to mention the alternative, as a curiosity.

Reply:
Applied in 592d0e2. This is more lines but less elements now. Thanks